// Author: Jiaying Guo

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.bigtable";
option java_outer_classname = "BigtableProto";
option objc_class_prefix = "BT";

package bigtable;

// The bigtable service definition.
service Bigtable {
  // put operation
  rpc put (PutRequest) returns (PutReply) {}

  // get operation
  rpc get (GetRequest) returns (GetReply) {}

  // cput operation
  rpc cput (CPutRequest) returns (CPutReply) {}

  // delete operation
  rpc table_delete (DelRequest) returns (DelReply) {}

  // getEmailList operation
  rpc list_all_emails_for_currUser (GetEmailListRequest) returns (GetEmailListReply) {}

  // getFileList operation
  rpc list_all_files_for_currUser (GetFileListRequest) returns (GetFileListReply) {}
}


message PutRequest {
  string created_time = 1;
  int32 size = 2;
  string file_name = 3;
  string file_type = 4;
  string file_from = 5;
  string row = 6;
  string data = 7;
}


message PutReply {
  bool ret = 1;
}

message GetRequest {
  string row = 1;
  string col = 2;
}


message GetReply {
  string content = 1;
}

message CPutRequest {
  string created_time = 1;
  int32 size = 2;
  string file_name = 3;
  string file_type = 4;
  string file_from = 5;
  string row = 6;
  string col = 7;
  string old_data = 8;
  string new_data = 9;
}


message CPutReply {
  bool ret = 1;
}

message DelRequest {
  string row = 1;
  string col = 2;
}

message DelReply {
  bool ret = 1;
}

message fileMetaData{
  string created_time = 1;
  int32 size = 2;
  string file_name = 3;
  string file_type = 4;
  string file_from = 5;
  string file_id = 6;
}

message GetEmailListRequest {
  string row = 1;
}

message GetEmailListReply {
  repeated fileMetaData metadata = 1;
}

message GetFileListRequest {
  string row = 1;
}

message GetFileListReply {
  repeated fileMetaData metadata= 1;
}








